Consul.Test/CatalogTest.cs

```diff
@@ -317,5 +317,102 @@ public async Task Catalog_ServicesForNodes()
+                    Node = "bar",
+                    Address = "192.168.10.11",
+                    Service = new AgentService
+                    {
+                        ID = "redis",
+                        Service = "redis",
+                        Port = 6379
+                    }
+                };
+                await client.Catalog.Register(terminatingGatewayEntry);
+
+                var terminatingGatewayConfigEntry = new TerminatingGatewayEntry
+                {
+                    Name = terminatingGatewayName,
+                    Services = new List<LinkedService>
+                    {
+                        new LinkedService
+                        {
+                            Name = "api",
+                            CAFile = "api/ca.crt",
+                            CertFile = "api/client.crt",
+                            KeyFile = "api/client.key",
+                            SNI = "my-domain"
+                        },
+                        new LinkedService
+                        {
+                            Name = "*",
+                            CAFile = "ca.crt",
+                            CertFile = "client.crt",
+                            KeyFile = "client.key",
+                            SNI = "my-alt-domain"
+                        }
+                    }
+                };
+                await client.Configuration.ApplyConfig(terminatingGatewayConfigEntry);
+
+                var ingressGatewayConfigEntry = new IngressGatewayEntry
+                {
+                    Name = ingressGatewayName,
+                    Listeners = new List<GatewayListener>
+                    {
+                        new GatewayListener
+                        {
+                            Port = 8888,
+                            Services = new List<ExternalService>
+                            {
+                                new ExternalService
+                                {
+                                    Name = "api"
+                                }
+                            }
+                        },
+                        new GatewayListener
+                        {
+                            Port = 9999,
+                            Services = new List<ExternalService>
+                            {
+                                new ExternalService
+                                {
+                                    Name = "redis"
+                                }
+                            }
+                        }
+                    }
+                };
+
+                await client.Configuration.ApplyConfig(ingressGatewayConfigEntry);
+
+                var gatewayServices = await client.Catalog.GatewayService(terminatingGatewayName);
+
+                Assert.NotEmpty(gatewayServices.Response);
+                Assert.Equal(ServiceKind.TerminatingGateway, gatewayServices.Response[0].GatewayKind);
```

Comment:
> It would also be good to check if other fields in the response are non-null, hence we know that the struct is correctly defined.

Just added the checks, you can see that we can't check all fields, cause some fields are just not present in the response for both gateways, and some just for ingress one

Reply:
We can add checks for port and protocol for the ingress one. Apart from that, it looks good.

Reply:
Sorry, just saw your message, added the assertions, tests went good

Reply:
Thanks!

Reply:
> Thanks!

Thanks for help with the PRs!